lib/src/execution/posix/execution.cc

```diff
@@ -199,226 +326,142 @@ namespace facter { namespace execution {
         // Create the pipes for stdin/stdout redirection
         int pipes[2];
         if (::pipe(pipes) < 0) {
-            throw execution_exception("failed to allocate pipe for stdin redirection.");
+            throw execution_exception((boost::format("failed to allocate pipe for stdin redirection: %1% (%2%).") % strerror(errno) % errno).str());
```

Comment:
If this is something we're going to do more, I like pulling it out into a helper function. I did so on Windows at https://github.com/puppetlabs/leatherman/blob/master/windows/src/system_error.cc.